string namespace_base
string gr_namespace
---
bool ack
